File: WowUp.Common/Models/Addons/AddonSearchResultFile.cs
using System;
using System.Collections.Generic;
using WowUp.Common.Enums;

namespace WowUp.Common.Models.Addons
{
    public class AddonSearchResultFile
    {
        public AddonChannelType ChannelType { get; set; }
        public string Version { get; set; }
        public IEnumerable<string> Folders { get; set; }
        public string GameVersion { get; set; }
        public string DownloadUrl { get; set; }
        public DateTime ReleaseDate { get; set; }
        public IEnumerable<AddonSearchResultDependency> Dependencies { get; set; }
    }
}

File: wowup-electron/src/app/models/wowup/addon-search-result-file.ts
import { AddonChannelType } from "../../../common/wowup/models";
import { AddonSearchResultDependency } from "./addon-search-result-dependency";

export interface AddonSearchResultFile {
  externalId?: string;
  channelType: AddonChannelType;
  version: string;
  folders: string[];
  gameVersion: string;
  downloadUrl: string;
  releaseDate: Date;
  dependencies?: AddonSearchResultDependency[];
  changelog?: string;
  title?: string;
  authors?: string;
}
